crates/editor/src/signature_help.rs

```diff
@@ -291,28 +327,171 @@ impl SignatureHelpState {
 }
 
 #[derive(Clone, Debug, PartialEq)]
+pub struct SignatureHelpData {
+    pub(crate) label: SharedString,
+    documentation: Option<Entity<Markdown>>,
+    highlights: Vec<(Range<usize>, HighlightStyle)>,
+}
+
+#[derive(Clone, Debug)]
 pub struct SignatureHelpPopover {
-    pub label: SharedString,
     pub style: TextStyle,
-    pub highlights: Vec<(Range<usize>, HighlightStyle)>,
+    pub signature: Vec<SignatureHelpData>,
+    pub current_signature: Rc<RefCell<usize>>,
```

Comment:
> `Rc<RefCell<`

What is the reason for this?
Every access in the code seem to be behind `&mut` on a parent anyway, even in the click handlers?